src/pineko/evolve.py

```diff
@@ -309,20 +372,49 @@ def evolve_grid(
         )
         for mur2 in mur2_grid
     ]
-    # We need to use ekompatibility in order to pass a dictionary to pineappl
-    fktable = grid.evolve(
-        ekompatibility.pineappl_layout(operators),
-        xir * xir * mur2_grid,
-        alphas_values,
-        "evol",
-        order_mask=order_mask,
-        xi=(xir, xif),
-    )
+
+    def prepare(operator, items):
+        (q2, _), op = items
+        info = PyOperatorSliceInfo(
+            fac0=operator.mu20,
+            x0=operator.bases.inputgrid.raw,
+            pids0=basis_rotation.evol_basis_pids,
+            fac1=q2,
+            x1=operator.bases.targetgrid.raw,
+            pids1=operator.bases.targetpids,
+            pid_basis=PyPidBasis.Pdg,
+        )
+        return (info, op.operator)
+
+    if operators_b is not None:
+        fktable = grid.evolve_with_slice_iter2(
+            map(lambda it: prepare(operators_a, it), operators_a.items()),
```

Comment:
this works and is efficient so we keep it as is -  but it is still a bit awkward ... maybe we should add a `map` function to the `EKO` object which calls a function `f(op, ep, self)` on every element ... opinions @giacomomagni @alecandido ?

Reply:
To me, even the built-in `map()` seems good enough.

Moreover, do you really need `self`?
I believe you should only need the *metadata*, not the full operator.